internal/controller/postgrescluster/pgmonitor.go

```diff
@@ -342,14 +309,25 @@ func addPGMonitorExporterToInstancePodSpec(
 	}
 	template.Spec.Volumes = append(template.Spec.Volumes, configVolume)
 
-	// add default exporter queries config volume
-	queriesVolume := corev1.Volume{Name: "exporter-queries"}
-	queriesVolume.ConfigMap = &corev1.ConfigMapVolumeSource{
-		LocalObjectReference: corev1.LocalObjectReference{
-			Name: exporterQueriesConfig.Name,
-		},
+	// The original "custom queries" ability allowed users to provide a file with custom queries;
+	// however, it would turn off the default queries. The new "custom queries" ability allows
+	// users to append custom queries to the default queries. This new behavior is feature gated.
+	// Therefore, we only want to add the default queries ConfigMap as a source for the
+	// "exporter-config" volume if the AppendCustomQueries feature gate is turned on OR if the
+	// user has not provided any custom configuration.
+	if util.DefaultMutableFeatureGate.Enabled(util.AppendCustomQueries) ||
+		cluster.Spec.Monitoring.PGMonitor.Exporter.Configuration == nil {
+
+		defaultConfigVolumeProjection := corev1.VolumeProjection{
+			ConfigMap: &corev1.ConfigMapProjection{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: exporterQueriesConfig.Name,
+				},
+			},
+		}
+		configVolume.VolumeSource.Projected.Sources = append(configVolume.VolumeSource.Projected.Sources,
```

Comment:
Let me just talk this out:
lines 301-310: we always have a `configVolume` that projects a volume from `cluster.Spec.Monitoring.PGMonitor.Exporter.Configuration` -- which might be nil, right? (And we've had this code for 2 years now.)

Now, lines 318-329:
* if you have the feature gate turned off & have provided no custom configuration, we add the default queries configmap to that projected volume -- the cluster now has both volumesources, but only default queries
* if you have the feature gate turned on & have provided no custom configuration, we add the default queries configmap to that projected volume -- the cluster now has both volumesources, but only default queries
* if you have the feature gate turned off & have provided custom configuration, we DON'T add the default queries configmap to that projected volume -- the cluster now has only custom queries (and one volumesource)
* if you have the feature gate turned on & have provided custom configuration, we add the default queries configmap to that projected volume -- the cluster now has both custom & default queries

OK, just wanted to type that out.

Reply:
:thinking: makes me wonder if we should be adding the empty volumeSources?

Reply:
We've been doing it for ~2 years, so I guess it's ok, but I don't love it.

Reply:
Yeah, if it's not too difficult, I think we should stop doing that. Are there downsides? Would it be possible for someone to get into a situation where we added no volumeSources?

Maybe add a backlog issue and we can plan this